sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -103,10 +103,11 @@ class JDBCOptions(
       s" and '$JDBC_NUM_PARTITIONS' are required.")
   val fetchSize = {
     val size = parameters.getOrElse(JDBC_BATCH_FETCH_SIZE, "0").toInt
-    require(size >= 0,
+    require(size >= 0 || size == Integer.MIN_VALUE,
```

Comment:
It does not make sense to the other data sources except MySQL. If needed, we can introduce a dialect specific checking APIs in [JdbcDialect.scala](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala)